Module 3/SQL_Assignment.txt

```diff
@@ -0,0 +1,1880 @@
+| 1002 |                3 |
+| 1007 |                2 |
+| 1004 |                1 |
+| 1003 |                1 |
++------+------------------+
+5 rows in set (0.00 sec)
+
+
+9) List the customer table if and only if one or more of the customers in the Customer table are located in SanJose. 
+
+
+mysql> SELECT cnum, cname, city, rating, snum
+    -> FROM customers
+    -> WHERE (SELECT COUNT(city)
+    ->          FROM customers
+    ->          WHERE city = 'SanJose') >= 1;
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2001 | Hoffman  | London  |    100 | 1001 |
+| 2002 | Giovanni | Rome    |    200 | 1003 |
+| 2003 | Liu      | SanJose |    200 | 1002 |
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2006 | Clemens  | London  |    100 | 1001 |
+| 2007 | Cisneros | SanJose |    300 | 1007 |
+| 2008 | Pererira | Rome    |    100 | 1004 |
++------+----------+---------+--------+------+
+7 rows in set (0.00 sec)
+
+
+10) Match salespeople to customers according to what city they live in. 
+
+mysql> SELECT salespeople.*
+    -> FROM salespeople
+    -> LEFT JOIN customers ON customers.city = salespeople.city;
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | Rifkin  | Barcelona |   15 |
+| 1008 | Fran    | London    |   25 |
+| 1008 | Fran    | London    |   25 |
++------+---------+-----------+------+
+10 rows in set (0.00 sec)
+
+
+
+11) Find all the customers in SanJose who have a rating above 200. 
+
+mysql> SELECT cname FROM customers WHERE city = 'SanJose' AND rating > 200;
++----------+
+| cname    |
++----------+
+| Cisneros |
++----------+
+1 row in set (0.00 sec)
+
+
+12) List the names and commissions of all salespeople in London. 
+
+mysql> SELECT sname, comm FROM salesperson WHERE city = 'London';
++--------+------+
+| sname  | comm |
++--------+------+
+| Peel   |   12 |
+| Motika |   11 |
+| Fran   |   25 |
++--------+------+
+3 rows in set (0.00 sec)
+
+
+13) List all the orders of Salesperson Motika from the orders table.
+
+mysql> SELECT onum, amt, odate, cnum, snum
+    -> FROM orders
+    -> WHERE snum = 1004;
++------+--------+----------+------+------+
+| onum | amt    | odate    | cnum | snum |
++------+--------+----------+------+------+
+| 3002 | 1900.1 | 10/03/90 | 2007 | 1004 |
++------+--------+----------+------+------+
+1 row in set (0.00 sec)
+
+
+14) Find all customers who booked orders on October 3. 
+
+mysql> SELECT customers.cname
+    -> FROM customers
+    -> LEFT JOIN orders ON customers.cnum = orders.cnum
+    -> WHERE orders.odate = '10/03/90';
++----------+
```

Comment:
We can have unique customer names.

Reply:
Yes ma'am I will make changes to the query.